Source/player.cpp

```diff
@@ -1772,19 +1760,7 @@ StartPlayerKill(int pnum, int earflag)
 		app_fatal("StartPlayerKill: illegal player %d", pnum);
 	}
 
-	if (plr[pnum]._pClass == HeroClass::Warrior) {
-		PlaySfxLoc(PS_DEAD, p->_px, p->_py); // BUGFIX: should use `PS_WARR71` like other classes
-	} else if (plr[pnum]._pClass == HeroClass::Rogue) {
-		PlaySfxLoc(PS_ROGUE71, p->_px, p->_py);
-	} else if (plr[pnum]._pClass == HeroClass::Sorcerer) {
-		PlaySfxLoc(PS_MAGE71, p->_px, p->_py);
-	} else if (plr[pnum]._pClass == HeroClass::Monk) {
-		PlaySfxLoc(PS_MONK71, p->_px, p->_py);
-	} else if (plr[pnum]._pClass == HeroClass::Bard) {
-		PlaySfxLoc(PS_ROGUE71, p->_px, p->_py);
-	} else if (plr[pnum]._pClass == HeroClass::Barbarian) {
-		PlaySfxLoc(PS_WARR71, p->_px, p->_py);
-	}
+	PlaySfxLoc(GetSoundByClass(pnum, 71), p->_px, p->_py);

```

Comment:
Would be nice to fix the issues in a separate change instead of hiding them inside a cleanup PR.